packages/jest-haste-map/src/index.js

```diff
@@ -273,6 +275,16 @@ class HasteMap {
         getPlatformExtension(module[H.PATH]) || H.GENERIC_PLATFORM;
       const existingModule = moduleMap[platform];
       if (existingModule && existingModule[H.PATH] !== module[H.PATH]) {
+        if (this._options.throwOnModuleCollision) {
+          throw new Error(
```

Comment:
it looks like we are duplicating the entire message twice now with the only difference being `error` and `warning`. Can we instead assign the message to a variable only once and do one of these:
* Instead of `warning` and `error`, use `message`.
* Use `${this._options.throwOnModuleCollision ? 'error' : 'message'}` :)